doc/protocols/casa.rst

```diff
@@ -0,0 +1,93 @@
+******************
+CASA Specification
+******************
+
+This document contains the specification of the CASA (Cryptographic 
+Agility for SCION ASes) protocol.
+
+Background
+==========
+In SCION, several protocols like DRKey and EPIC require different 
+entities (ASes, hosts) to agree on the same cryptographic 
+algorithms like PRFs and MACs.
+
+The simplest solution would be to globally define the algorithm 
+(e.g. AES-based CMAC) that is used by such a protocol. If an 
+algorithm becomes obsolete, the protocol can be redefined using a 
+new Path Type, and another algorithm may be introduced. This 
+solution assumes that all the ASes in the whole Internet can agree 
+on the same cryptographic algorithms.
+
+However, ASes may want to be more flexible in the choice of the 
+algorithms they support. A more sophistic approach would be that 
+each AS promotes the subset of algorithms (from some globally 
+defined set of algorithms) that it supports, similarly to the 
+exchange of cipher suites in TLS handshakes. This means however, 
+that the data plane packets need to introduce an additional field 
+for each AS, which identifies the selected algorithm. For DRKey and 
+EPIC, border routers would need to potentially support many 
+different algorithms, which hinders scalability and performance.
+
+CASA can provide most of the advantages of both those two options, 
+while avoiding their drawbacks: it is a tradeoff between simplicity 
+and agility.
+
+Specification
+=============
+
+Description
+-----------
+In CASA, every AS chooses exactly **one algorithm** per algorithm 
```

Comment:
From my point of view it would make sense to have them added directly to this document.
I added an additional section to address the concrete algorithms and categories in the future.

Reply:
:+1: